examples/java/src/main/proto/TimeSeriesData.proto

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+syntax = "proto3";
+
+package protos;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+option java_outer_classname = "TimeSeriesData";
+option java_multiple_files = false;
+option java_package = "org.apache.beam.examples.timeseries.protos";
+
+//	The TimeSeriesData proto type will be used as the internal representation of all timeseries within the the TS
```

Comment:
Is it a requirement for this to be a Proto (e.g. if you're planning to add cross-language support)? If not, then I personally would suggest to using raw Java classes instead. I imagine not everyone is familiar with protos, plus you can have more customization options for Java classes, e.g. introduce utility and/or factory methods.

Reply:
The expectation is that downstream the data can be used in other languages ( mainly python). 

Utility class for each of the main protos. In similar style to Timestamp with protobuff which has utils in Timestamps. Each of the protos have a (s) util class.